components/eam/src/physics/cam/hetfrz_classnuc.F90

```diff
@@ -310,272 +250,467 @@ subroutine hetfrz_classnuc_calc( &
+   real(r8), parameter :: hplanck = 6.63e-34_r8   ! planck constant
+   real(r8), parameter :: rhplanck = 1._r8/hplanck
+
+   real(r8) :: rgimm_bc                           ! critical germ radius for bc immersion freezing [m]
+   real(r8) :: rgimm_dust_a1, rgimm_dust_a3       ! critical germ radius for dust immersion freezing [m]
+   real(r8) :: dg0imm_bc                          ! homogeneous energy for germ formation [J]
+   real(r8) :: dg0imm_dust_a1, dg0imm_dust_a3     ! homogeneous energy for germ formation [J]
+   real(r8) :: Aimm_bc                            ! prefactor for bc [m^-2 s^-1]
+   real(r8) :: Aimm_dust_a1, Aimm_dust_a3         ! prefactor for dust [m^-2 s^-1]
+   real(r8) :: f_imm_bc                           ! form factor for bc 
+   real(r8) :: f_imm_dust_a1, f_imm_dust_a3       ! form factor for dust
+   real(r8) :: Jimm_bc                            ! immersion nucleation rate for bc [/s]
+   real(r8) :: Jimm_dust_a1, Jimm_dust_a3         ! immersion nucleation rate for dust [/s]
+   real(r8) :: ibin
+
+   real(r8) :: dim_f_imm_dust_a1(pdf_n_theta), dim_f_imm_dust_a3(pdf_n_theta)
+   real(r8) :: dim_Jimm_dust_a1(pdf_n_theta), dim_Jimm_dust_a3(pdf_n_theta)
+   real(r8) :: sum_imm_dust_a1, sum_imm_dust_a3
+
+   
+
    ! form factor
    ! only consider flat surfaces due to uncertainty of curved surfaces
+   f_imm_bc = get_form_factor(theta_imm_bc*pi/180._r8)
+
+   dim_f_imm_dust_a1 = 0.0_r8
+   dim_f_imm_dust_a3 = 0.0_r8
+   do ibin = itheta_bin_beg, itheta_bin_end
+      dim_f_imm_dust_a1(ibin) = get_form_factor(dim_theta(ibin))
+
+      dim_f_imm_dust_a3(ibin) = get_form_factor(dim_theta(ibin))
+   enddo
 
-   m = COS(theta_imm_bc*pi/180._r8)
-   f_imm_bc = (2+m)*(1-m)**2/4._r8
-   if (.not. pdf_imm_in) then
-      m = COS(theta_imm_dust*pi/180._r8)
-      f_imm_dust_a1 = (2+m)*(1-m)**2/4._r8
-
-      m = COS(theta_imm_dust*pi/180._r8)
-      f_imm_dust_a3 = (2+m)*(1-m)**2/4._r8
-   else 
-      dim_f_imm_dust_a1 = 0.0_r8
-      dim_f_imm_dust_a3 = 0.0_r8
-      do i = i1,i2
-         m = cos(dim_theta(i))
-         dim_f_imm_dust_a1(i) = (2+m)*(1-m)**2/4._r8
-
-         m = cos(dim_theta(i))
-         dim_f_imm_dust_a3(i) = (2+m)*(1-m)**2/4._r8
-      end do
-   end if
 
    ! homogeneous energy of germ formation
-   dg0imm_bc = 4*pi/3._r8*sigma_iw*rgimm_bc**2
+   dg0imm_bc      = 4*pi/3._r8*sigma_iw*rgimm_bc**2
    dg0imm_dust_a1 = 4*pi/3._r8*sigma_iw*rgimm_dust_a1**2
    dg0imm_dust_a3 = 4*pi/3._r8*sigma_iw*rgimm_dust_a3**2
 
    ! prefactor
-   Aimm_bc = n1*((vwice*rhplanck)/(rgimm_bc**3)*SQRT(3._r8/pi*kboltz*T*dg0imm_bc))
-   Aimm_dust_a1 = n1*((vwice*rhplanck)/(rgimm_dust_a1**3)*SQRT(3._r8/pi*kboltz*T*dg0imm_dust_a1))
-   Aimm_dust_a3 = n1*((vwice*rhplanck)/(rgimm_dust_a3**3)*SQRT(3._r8/pi*kboltz*T*dg0imm_dust_a3))
+   Aimm_bc = n1*((vwice*rhplanck)/(rgimm_bc**3)*SQRT(3._r8/pi*kboltz*temperature*dg0imm_bc))
```

Comment:
if you changed the capitals to lower cases in other places, maybe also change `SQRT` here to `sqrt` (and `EXP` a few lines later)

Reply:
changed all capital functions to lower cases